app/javascript/lib/scp-api.js

```diff
@@ -119,34 +153,55 @@ export async function fetchFacetFilters(facet, query, mock=false) {
     queryString = `_${facet}_${query}`
   }
 
+  logFilterSearch(facet, query)
+
   const pathAndQueryString = `/search/facet_filters${queryString}`
 
-  return await scpApi(pathAndQueryString, defaultInit, mock)
+  const filters = await scpApi(pathAndQueryString, defaultInit, mock)
+
+  mapFilters(filters)
+
+  return filters
 }
 
 /**
  * Returns a list of matching studies given a keyword and facets
  *
  * Docs: https:///singlecell.broadinstitute.org/single_cell/api/swagger_docs/v1#!/Search/search_facet_filters_path
  *
- * @param {type} Type of query to perform (study- or cell-based)
- * @param {terms} User-supplied query string
- * @param {facets} User-supplied list facets and filters
+ * @param {String} type Type of query to perform (study- or cell-based)
+ * @param {String} terms User-supplied query string
+ * @param {Object} facets User-supplied list facets and filters
+ * @param {Integer} page User-supplied list facets and filters
+ * @param {Boolean} mock Whether to use mock data
  * @returns {Promise} Promise object containing camel-cased data from API
  *
  * @example
  *
  * fetchSearch('study', 'tuberculosis');
  */
-export async function fetchSearch(type, terms, facets, page, mock=false) {
-  const searchPathAndQueryString = `/search?${buildSearchQueryString(type, terms, facets, page)}`
-  return await scpApi(searchPathAndQueryString, defaultInit, mock)
+export async function fetchSearch(
+  type, terms, facets, page, mock=false
+) {
+  const path = `/search?${buildSearchQueryString(type, terms, facets, page)}`
+
+  logSearch(type, terms, facets, page)
+
+  return await scpApi(path, defaultInit, mock)
 }
 
+/**
+ * Constructs query string used for /search REST API endpoint
+ */
 export function buildSearchQueryString(type, terms, facets, page) {
-  return `type=${type}&terms=${terms}&facets=${buildFacetQueryString(facets)}&page=${page ? page : 1}`
+  const facetsParam = buildFacetQueryString(facets)
+  const pageParam = page ? page : 1
+  return `type=${type}&terms=${terms}&facets=${facetsParam}&page=${pageParam}`
 }
 
+/**
+ * Serializes "facets" URL parameter for /search API endpoint
+ */
```

Comment:
I wouldn't mind a convention where simple one-line function comments like these don't use block-comment syntax, just so that we can see more of a class at once.  But I can see both sides

Reply:
Agreed, if we can get:

1. IDEs to show such docstrings in imported references, and
2. The linter to not throw warnings for non-block docstrings while still warning about functions without docstrings

Reply:
Done in f7640f5!